core/src/View/Kokkos_ViewCtor.hpp

```diff
@@ -213,11 +211,12 @@ struct ViewCtorProp : public ViewCtorProp<void, P>... {
   using execution_space = typename var_execution_space::type;
   using pointer_type    = typename var_pointer::type;
 
-  /*  Copy from a matching argument list.
-   *  Requires  std::is_same< P , ViewCtorProp< void , Args >::value ...
-   */
+  ViewCtorProp() = default;
```

Comment:
Why is it a good idea to create a default-constructed object (possibly with invalid values)?

Reply:
I thin k it was possible before this PR already...

But if you ask me, I'd say a defaulted view property has no label, and basically all the rest is defaulted too (memory space, execution space, initialization and so on).

Reply:
> I thin k it was possible before this PR already...

No, it wasn't. You would get a mismatch in the size of the `Args..` and `P...` parameter packs.

Reply:
> But if you ask me, I'd say a defaulted view property has no label, and basically all the rest is defaulted too (memory space, execution space, initialization and so on).

Sure it's the same as not using a `ViewCtorProp` argument  in the respective API calls. The point is that you would specify template arguments for which you would explicitly want to set a non-default value and so a default constructor somewhat defeats that purpose.

Reply:
I might have something wrong in my setting, but this compiles (current develop branch):
```c++
TEST(try, it)
{
    using ctor_t = Kokkos::Impl::ViewCtorProp<void, std::string>;
    static_assert(std::is_default_constructible_v<ctor_t>);
}
```

So it seems that it's already default constructible, and this PR does not change the current behavior.

Reply:
We are considerd about
```C++
using ctor_t = Kokkos::Impl::ViewCtorProp<std::string>;
static_assert(std::is_default_constructible_v<ctor_t>);
```
which would also work but
```C++
ctor_t bla;
```
fails. `is_default_constructible_v` only detects if there is a default constructor. It doesn't detect if it's instantiable (if it's a variadic template), also see https://godbolt.org/z/b4dMb91Pz.

Reply:
Since this discussion, the PR has been reworked. The PR now leaves unchanged what is default constructible and what is not. In short, the base classes have an explicitly defaulted constructor. The derived class is default constructible if it is empty. But it is not default constructible if it holds one or more properties.